samples/mpsl/timeslot/src/main.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <kernel.h>
+#include <console.h>
+#include <string.h>
+#include <misc/printk.h>
+#include <zephyr/types.h>
+#include <irq.h>
+#include <dk_buttons_and_leds.h>
+
+#include <mpsl_timeslot.h>
+#include <mpsl.h>
+
+#define RUN_STATUS_LED               DK_LED1
+#define TIMESLOT_STATUS_LED          DK_LED2
+
+#define RUN_STATUS_BLINK_INTERVAL_US (1000)
+#define TIMESLOT_REQUEST_DISTANCE_US (1000000)
+#define TIMESLOT_LENGTH_US           (200)
+
+#define MPSL_IRQ_LOW_PRIO            (4)
+#define STACKSIZE                    CONFIG_IDLE_STACK_SIZE
+#define THREAD_PRIORITY              K_LOWEST_APPLICATION_THREAD_PRIO
+
+static volatile int timeslot_counter;
+static bool request_in_cb = true;
+
+/* Timeslot requests */
+static mpsl_timeslot_request_t timeslot_request_earliest = {
+	.request_type = MPSL_TIMESLOT_REQ_TYPE_EARLIEST,
+	.params.earliest.hfclk = MPSL_TIMESLOT_HFCLK_CFG_NO_GUARANTEE,
+	.params.earliest.priority = MPSL_TIMESLOT_PRIORITY_NORMAL,
+	.params.earliest.length_us = TIMESLOT_LENGTH_US,
+	.params.earliest.timeout_us = 1000000
+};
+static mpsl_timeslot_request_t timeslot_request_normal = {
+	.request_type = MPSL_TIMESLOT_REQ_TYPE_NORMAL,
+	.params.normal.hfclk = MPSL_TIMESLOT_HFCLK_CFG_NO_GUARANTEE,
+	.params.normal.priority = MPSL_TIMESLOT_PRIORITY_NORMAL,
+	.params.normal.distance_us = TIMESLOT_REQUEST_DISTANCE_US,
+	.params.normal.length_us = TIMESLOT_LENGTH_US
+};
+
+static mpsl_timeslot_signal_return_param_t signal_callback_return_param;
+
+/* FIFO used for printing the signal type from timeslot callback */
+K_FIFO_DEFINE(cb_fifo);
```

Comment:
I think a message queue might be a better fit here. It'll remove the need for the malloc call in the ISR, and it wraps the buffer handling in the queue object, so it'll look a bit cleaner on both sides.